ACPs/118-warp-signature-request/README.md

```diff
@@ -0,0 +1,88 @@
+
+    ```protobuf
+    message SignatureRequest struct {
+        bytes data = 1;
+        bytes justification = 2;
+    }
+    ```
+
+- `SignatureResponse` is the corresponding `AppResponse` type that returns the requested signature. 
+
+    ```protobuf
+    message SignatureResponse {
+        bytes signature = 1;
+    }
+    ```
+
+For each of these types, VMs must implement corresponding `AppRequest` and `AppResponse` handlers to adhere to the interface.
+
+## Use Cases
+
+Generally speaking, `SignatureRequest` can be used to request a signature over a Warp message by serializing the unsigned Warp message into `data`, and populating `justification` as needed.
+
+### Sign a known Warp Message
+
+Subnet EVM and Coreth store messages that have been seen (i.e. on-chain message sent through the [Warp Precompile](https://github.com/ava-labs/subnet-evm/tree/v0.6.7/precompile/contracts/warp) and [off-chain](https://github.com/ava-labs/subnet-evm/blob/v0.6.7/plugin/evm/config.go#L226) Warp messages) such that a signature over that message can be provided on request. `SignatureRequest` can be used for this case by specifying the Warp message in `data`. The queried node may then look up the Warp message in its database and return the signature. In this case, `justification` is unneeded.
+
+### Attest to an on-chain event
+
+Subnet EVM and Coreth also support attesting to block hashes via Warp, by serving signature requests made using the following `AppRequest` type:
+
+```
+type BlockSignatureRequest struct {
+	BlockID ids.ID
+}
+```
+
+`SignatureRequest` can achieve this by specifying an unsigned Warp message with the `BlockID` as the payload, and serializing that message into `data`. `justification` may optionally be used to provide additional context, such as a the block height of the given block ID.
+
+### Confirm that an event did not occur
+
+With [ACP-77](https://github.com/avalanche-foundation/ACPs/tree/main/ACPs/77-reinventing-subnets), Subnets will have the ability to manage their own validator sets. The Warp message payload contained in a `RegisterSubnetValidatorTx` includes an `expiry`, after which the specified validation ID becomes invalid. The Subnet needs to know that this validation ID is expired so that it can keep its locally tracked validator set in sync with the P-Chain. We also assume that the P-Chain will not persist expired or invalid validation IDs. 
+
+We can use `SignatureRequest` to construct a Warp message attesting that the validation ID expired. We do so by serializing an unsigned Warp message containing the validation ID into `data`, and providing additional metadata in `justification` for the P-Chain to reconstruct the expired validation ID.
+
+## Security Considerations
+TODO
```

Comment:
Should complete this section before merging even in draft imo

Reply:
Agreed. Think the only consideration here is that VMs implementing the proposed standard interface need to consider how to verify what data they should be willing to sign, and if the implementation is DOSable. The implementation is left to the VMs to provide themselves though.

Reply:
Added a list of security considerations that VM implementers should take into account.